### PR TITLE
APEXMALHAR-2154 Update the Kafka Input Operator to use CheckpointNotificationListener

### PR DESCRIPTION
**Pull request by chaithu14**

### CODE REVIEW

**Comment by sandeshh** [comment]
Should this be done for every operators in Malhar?

File: kafka/src/main/java/org/apache/apex/malhar/kafka/AbstractKafkaInputOperator.java [hunk 1]
```diff
@@ -191,6 +191,12 @@ public void checkpointed(long l)
   }
 
   @Override
+  public void beforeCheckpoint(long windowId)
```
Comment (tweise) [inline kafka/src/main/java/org/apache/apex/malhar/kafka/AbstractKafkaInputOperator.java:194]: @chaithu14 what is the purpose of this change? Why add an interface with no implementation?

Comment (chaithu14) [reply kafka/src/main/java/org/apache/apex/malhar/kafka/AbstractKafkaInputOperator.java:194]: JIRA represents to update the checkpointlistner. In this operator, there is no functionality in checkpoint.
